gk/main.c

```diff
@@ -1015,10 +1143,13 @@ process_pkts_front(uint8_t port_front, uint8_t port_back,
 
 	/* XXX Do something better here! For now, free any unsent packets. */
 	if (unlikely(num_tx_succ < num_tx)) {
-		for (i = num_tx_succ; i < num_tx; i++)
+		for (i = num_tx_succ; i < num_tx; i++) {
+			tx_bytes -= tx_bufs[i]->pkt_len;
 			drop_packet(tx_bufs[i]);
+		}
 	}
 
+	rte_red_mark_queue_empty(&instance->red, tx_bytes);
```

Comment:
I agree -- the only way RED would be useful is if `GATEKEEPER_MAX_PKT_BURST` is greater than the minimum threshold.

My reasoning before for accepting this condition is because without having done experiments to determine `GATEKEEPER_MAX_PKT_BURST` and the RED minimum threshold, it seemed possible that we might settle on values for these such that `GATEKEEPER_MAX_PKT_BURST` is greater than the minimum threshold and therefore RED is useful.

However, now I realize that if `GATEKEEPER_MAX_PKT_BURST` were greater than the RED minimum threshold, it would imply that `GATEKEEPER_MAX_PKT_BURST` is already too much to handle (and would cause congestion).

So I am leaning toward just having Gatekeeper servers maintain the ECN field. The good thing is that this change would greatly simplify this patch, not make it more complicated. Should I make these changes?